### PR TITLE
Add get censored words feature

### PR DESCRIPTION
**Pull request by emso-c**
Provided a solution for the issue #34. Sorry I kind of messed up with branches so this commit is merged with the other PR I created (#35).

Again, it doesn't break anything and can only be used if `get_censored_words` is `True`

It basically returns a Tuple of `(str, list)` with the `str` being the original censored text and the `list` being the list of censored words.

### Usage:
```
from better_profanity import profanity

if __name__ == "__main__":
    profanity.load_censor_words()
    
    text = "test fucking shit"
    censored_text, censored_words = profanity.censor(text, get_censored_words=True)
    print(censored_words)
    # ['fucking', 'shit']
```

### CODE REVIEW

File: better_profanity/better_profanity.py [hunk 1]
```diff
@@ -53,7 +54,7 @@ def __init__(self, words=None):
 
     ## PUBLIC ##
 
-    def censor(self, text, censor_char="*"):
+    def censor(self, text, censor_char="*", middle_only=False, get_censored_words=False):
```
Comment (snguyenthanh) [inline better_profanity/better_profanity.py:57]: I think it would be better to have a separate `get_censored_words(text)` function, as it is not obviously clear the returned result of `censor(get_censored_words=True)` is. You could share some/most of the code with `censor` function.

**Review by snguyenthanh — CHANGES_REQUESTED** [review]
It would be great if you could write some tests for the functions you added as well.

File: better_profanity/utils.py [hunk 2]
```diff
@@ -18,8 +18,14 @@ def read_wordlist(filename: str):
                 yield row
 
 
-def get_replacement_for_swear_word(censor_char):
-    return censor_char * 4
+def get_replacement_for_swear_word(censor_char, n=4):
```
Comment (snguyenthanh) [inline better_profanity/utils.py:21]: It's just my personal preference: could you replace `n` with a more detailed variable ? It's quite unclear what `n` is when we call `get_replacement_for_swear_word("-", n=2)`.

Comment (emso-c) [reply better_profanity/utils.py:21]: Sure.

File: better_profanity/better_profanity.py (hunk 1 above)
Comment (snguyenthanh) [reply better_profanity/better_profanity.py:57]: That's great. You are right that it would be better to not repeat the same code. But we could also create a function that is used by both `censor` and `get_censored_words` to determine which words are profane.

**Comment by emso-c** [comment]
Separated the functions as you said but while writing unit tests and testing edge cases, I just realized that the current `_hide_swear_words` function merges multiple swear words into one when they're next to each other. Though both functionalities work well for single unseparated swear words, they don't behave well in these kind of situations.

Example with get_censored_words:
```python
bad_text = "Dude, I hate shit. Fuck bullshit."
profanity.get_censored_words(bad_text)
>>>['shit', 'bullshit']
# It completely ignored "Fuck" since they're merged

bad_text = "That wh0re gave m3 a very good H@nD j0b."
profanity.get_censored_words(bad_text)
>>>['wh0re', 'H@nD']
# It didn't include "j0b" since they're separated with space
```
Example with middle_only (same issues):
```python
bad_text = "Dude, I hate shit. Fuck bullshit."
profanity.censor(bad_text, middle_only=True)
>>>"Dude, I hate s**t b******t."
# It completely ignored "Fuck" since they're merged

bad_text = "That wh0re gave m3 a very good H@nD j0b."
profanity.censor(bad_text, middle_only=True)
>>>"That w***e gave m3 a very good H**D."
# It didn't include "j0b" since they're separated with space
```

<hr>
<br>

To solve that, I simply put a check before merging swear words (only merge if: `not (get_censored_words or middle_only)`). The results are better and they pass all other unit tests, but bit of a coverage your method provided has disappeared. Which means it'll detect less swear words and it might result in inconsistencies.

Example with get_censored_words:
```python
bad_text = "Dude, I hate shit. Fuck bullshit."
profanity.get_censored_words(bad_text)
>>>['shit', 'Fuck', 'bullshit']

bad_text = "That wh0re gave m3 a very good H@nD j0b."
profanity.get_censored_words(bad_text)
>>>['wh0re']
# It didn't include "H@nD j0b"
```
Example with middle_only (same issues):
```python
bad_text = "Dude, I hate shit. Fuck bullshit."
profanity.censor(bad_text, middle_only=True)
>>>"Dude, I hate s**t. F**k b******t."


bad_text = "That wh0re gave m3 a very good H@nD j0b."
profanity.censor(bad_text, middle_only=True)
>>>"That w***e gave m3 a very good H@nD j0b."
# It didn't include "H@nD j0b"
``` 
<hr>
<br>
Maybe we should just follow this method and warn users of these possible issues? I think it's a pretty mild edge case anyway, but it's up to you.
<br><br>
